x/chainlink/keeper/keeper.go

```diff
@@ -468,3 +471,36 @@ func (k Keeper) RequestNewRound(ctx sdk.Context, requestNewRound *types.MsgReque
 
 	return ctx.BlockHeight(), ctx.TxBytes(), nil
 }
+
+func (k Keeper) AddAccount(ctx sdk.Context, acc *types.MsgAccount) (int64, []byte) {
+	accStore := ctx.KVStore(k.accountStoreKey)
+
+	a := k.cdc.MustMarshalBinaryBare(acc)
+
+	accStore.Set(types.GetAccountKey(acc.GetSubmitter().String()), a)
+
+	return ctx.BlockHeight(), ctx.TxBytes()
+}
+
+func (k Keeper) EditAccount(ctx sdk.Context, acc *types.MsgEditAccount) (int64, []byte, error) {
+	accStore := ctx.KVStore(k.accountStoreKey)
+
+	a := k.cdc.MustMarshalBinaryBare(acc)
+
+	accStore.Set(types.GetAccountKey(acc.GetSubmitter().String()), a)
+
+	return ctx.BlockHeight(), ctx.TxBytes(), nil
+}
```

Comment:
Look like you are overwriting the existing `types.MsgAccount` in the store with `types.MsgEditAccount` that do not contain the same fields.
I think you should access the current content in the store based on the submitter as a key, unmarshal the result to `types.MsgAccount`, modify the piggy address and store it back (also check if the result didn't exist).

Reply:
you are absolutely correct... thanks for catching that!!!!